src/linear_algebra/free_module/pid.lean

```diff
@@ -608,6 +608,10 @@ begin
   simp [hi]
 end
 
+instance (F : Type u) [comm_ring F] [algebra F R] (b : basis ι R S) {I : ideal S} (hI : I ≠ ⊥) :
+  Π i, module F (R ⧸ ideal.span ({I.smith_coeffs b hI i} : set R)) :=
```

Comment:
Sorry for the late reply; I forgot about the PR ... What I had in mind was
```
instance submodule.quotient.module'.pi {R M} [ring R] [add_comm_group M] [module R M]
  {S} [semiring S] [has_smul S R] [module S M] [is_scalar_tower S R M] {ι} (P : ι → submodule R M) :
  Π i, module S (M ⧸ P i) := by apply_instance
```
but actually it does nothing (and `quotient_equiv_direct_sum` doesn't work with it), because apparently the problem isn't about pi instances: if you replace this by
```
instance (i) : module F (R ⧸ span ({I.smith_coeffs b hI i} : set R)) := by apply_instance
```
then `quotient_equiv_direct_sum` still works! It's weird we need to add a shortcut instance that is already found by typeclass inference.